src/main/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpoint.java

```diff
@@ -284,15 +284,18 @@ private void createNewEventForAccessCodeAuthAttempt(final Case targetCase) throw
   }
 
   /**
-   * the GET endpoint to find case events by case id
+   * the GET endpoint to find case events, with optional timestamps by case id
    *
    * @param caseId to find by
-   * @return the case events found
+   * @param category type to collect case event timestamp
+   * @return the case events found, with optional timestamps (category required for this)
    * @throws CTPException something went wrong
    */
   @RequestMapping(value = "/{caseId}/events", method = RequestMethod.GET)
   public ResponseEntity<List<CaseEventDTO>> findCaseEventsByCaseId(
-      @PathVariable("caseId") final UUID caseId) throws CTPException {
+      @PathVariable("caseId") final UUID caseId,
+      @RequestParam(value = "category", required = false) final List<String> categories)
```

Comment:
I still think this should be category name rather than string. Will we want a list or just one category?

Reply:
So, tried that, it throws a 500 instead of a 400 when the category name doesnt exist, which is wrong behaviour. To get around that needs to be passed as a string so just keeping it this way for now